unidist/core/backends/mpi/core/shared_object_store.py

```diff
@@ -154,30 +151,90 @@ def _allocate_shared_memory(self):
 
+        if shared_object_store_memory is not None:
+            if shared_service_memory is not None:
+                self.shared_memory_size = shared_object_store_memory
+                self.service_memory_size = shared_service_memory
+            else:
+                self.shared_memory_size = shared_object_store_memory
+                # To avoid division by 0
+                if MpiSharedObjectStoreThreshold.get() > 0:
+                    self.service_memory_size = min(
+                        # allowd memory for serivce buffer
+                        allowed_memory_size - self.shared_memory_size,
+                        # maximum amount of memory required for the service buffer
+                        (self.shared_memory_size // MpiSharedObjectStoreThreshold.get())
+                        * (self.INFO_SIZE * MPI.LONG.size),
+                    )
+                else:
+                    self.service_memory_size = (
+                        allowed_memory_size - self.shared_memory_size
+                    )
+        else:
+            if shared_service_memory is not None:
+                self.service_memory_size = shared_service_memory
+                self.shared_memory_size = allowed_memory_size - self.service_memory_size
+            else:
+                A = allowed_memory_size
+                B = MpiSharedObjectStoreThreshold.get()
+                C = self.INFO_SIZE * MPI.LONG.size
+                # "x" is shared_memory_size
+                # "y" is service_memory_size
+
+                # requirements:
+                # x + y = A
+                # y = min[ (x/B) * C, 0.01 * A ]
+
+                # calculation results:
+                # if B > 99 * C:
+                # x = (A * B) / (B + C)
+                # y = (A * C) / (B + C)
+                # else:
+                # x = 0.99 * A
+                # y = 0.01 * A
+
+                if B > 99 * C:
+                    self.shared_memory_size = (A * B) // (B + C)
+                    self.service_memory_size = (A * C) // (B + C)
+                else:
+                    self.shared_memory_size = int(0.99 * A)
+                    self.service_memory_size = int(0.01 * A)
+
+        if self.shared_memory_size > allowed_memory_size:
+            raise ValueError(
+                "Memory for shared object storage cannot be allocated because the value set to `MpiSharedObjectStoreMemory` exceeds the available memory."
+            )
+
+        if self.service_memory_size > allowed_memory_size:
+            raise ValueError(
+                "Memory for shared service storage cannot be allocated because the value set to `MpiSharedServiceMemory` exceeds the available memory."
+            )
+
+        if self.shared_memory_size + self.service_memory_size > allowed_memory_size:
+            raise ValueError(
+                "The sum of the `MpiSharedObjectStoreMemory` and `MpiSharedServiceMemory` values is greater than the amount of memory that exists."
```

Comment:
```suggestion
                "The sum of the `MpiSharedObjectStoreMemory` and `MpiSharedServiceMemory` values is greater than the available amount of memory."
```
Put this message in multiple lines but not in one.